shared-components/tl-tournament-card.js

```diff
@@ -251,26 +208,51 @@ export class TournamentCard extends LitElement {
                 return html` <h4>loading</h4> `;
             case State.LOADED:
                 return html`
-                    <h4>Title: ${this.tournament.name}</h4>
+                    <img src=${this.tournament.eventImage} />
+                    <h4 class="event-title">${this.tournament.name}</h4>
                     <!-- FIXME: change the value of the substring to match official given value -->
-                    <div class="paragraph-large">Short description: ${this.tournament.description.substring(0, 30)}</div>
-                    <!-- FIXME: show dates like a countdown -->
-                    <div class="paragraph-medium">Event dates: ${this._formatDates()}</div>
-                    <!-- FIXME: show real prize amount -->
-                    <!-- FIXME: use the localization API to generate the plural of the currency symbol -->
-                    <div class="paragraph-large">Prize money: ${this.tournament.fee}&nbsp;${this.tournament.currency.symbol}s</div>
-                    <mwc-button class="details-button button-label-large" label="Details" outlined @click="${(event) => this._handleDetailsButton(event)}"></mwc-button>
+                    <div class="paragraph-large event-text">
+                        ${this.tournament.description.substring(0, 30)}
+                    </div>
+                    <tl-tournament-countdown
+                        class="counter"
+                        date-start="${this.tournament.dateStart}"
+                        date-end="${this.tournament.dateEnd}"
+                    ></tl-tournament-countdown>
+                    <div class="paragraph-large event-prize">
+                        Prize money <span class="event-prize-amount">${this._currentCurrency}</span>
+                    </div>
+                    <mwc-button
+                        class="details-button button-label-large"
+                        label="Details"
+                        outlined
+                        @click="${(event) => this._handleDetailsButton(event)}"
+                    ></mwc-button>
                 `;
             case State.ERROR:
                 return html` <h4>error</h4> `;
         }
     }
 
-    willUpdate() {
+    async willUpdate() {
         if (this.entityId !== null && this._state === State.EMPTY) {
             this._fetchTournamentInformation(this.entityId);
         }
+        if (this.tournament && !this._currentCurrency) {
+            this._convertCurrency();
+        }
+    }
+
+    //do this cause mwc-button has fixed height and issue still open https://github.com/material-components/material-web/issues/81
+    async updated() {
```

Comment:
Same comment on `updated()` having to be non blocking method.

This is too much of a hack! Instead of trying to rely on a private API (accessing the button inside the shadow DOM of the `<mwc-button/>` is a really bad idea... I would prefer a suggestion of cloning the component so we can have the button height that rely on a public variable, so we can style it from outside as we do for background color, for example.